src/CaNNOLeS.jl

```diff
@@ -26,19 +27,64 @@ include("solver_types.jl")
 
 Implementation of a solver for Nonlinear Least Squares with nonlinear constraints.
 
+``
   min   f(x) = ¹/₂‖F(x)‖²   s.t.  c(x) = 0
+``
 
 For advanced usage, first define a `CaNNOLeSSolver` to preallocate the memory used in the algorithm, and then call `solve!`:
-  solver = CaNNOLeSSolver(nls)
-  solve!(solver, nls; kwargs...)
 
-or even pre-allocate the output:
+    solver = CaNNOLeSSolver(nls)
+    solve!(solver, nls; kwargs...)
 
-  stats = GenericExecutionStats(nls)
-  solve!(solver, nls, stats; kwargs...)
+or even pre-allocate the output:
 
-Input:
-- `nls :: AbstractNLSModel`: Nonlinear least-squares model created using `NLPModels`.
+    stats = GenericExecutionStats(nls)
+    solve!(solver, nls, stats; kwargs...)
+
+# Arguments
+- `nls :: AbstractNLSModel`: nonlinear least-squares model created using `NLPModels`.
+
+# Keyword arguments 
+- `x::AbstractVector = nls.meta.x0`: the initial guess;
+- `λ::AbstractVector = eltype(x)[]`: the initial Lagrange multiplier;
+- `method::Symbol = :Newton`: available methods `:Newton, :LM, :Newton_noFHess`, and `:Newton_vanishing`;
+- `linsolve::Symbol = :ma57`: solver to compute LDLt factorization. Available methods are: `:ma57`, `:ldlfactorizations`;
+- `max_f::Real = 100000`: maximum number of evaluations computed by `sum_counters(nls)`;
+- `max_time::Float64 = 30.0`: maximum time limit in seconds;
+- `max_inner::Int = 10000`: maximum number of inner iterations;
+- `ϵtol::Real = √eps(eltype(x))`: stopping tolerance;
+- `verbose::Int = 0`: if > 0, display iteration details every `verbose` iteration;
+- `check_small_residual::Bool = true`: if `true`, stop whenever ``‖F(x)‖₂² ≤ ϵtol`` and ``‖c(xᵏ)‖∞ ≤ √ϵtol``;
+- `always_accept_extrapolation::Bool = false`: if `true`, run even if the extrapolation step fails;
+- `δdec::Real = eltype(x)(0.1)`: reducing factor on the parameter `δ`.
+
+The algorithm stops when ``‖c(xᵏ)‖∞ ≤ ϵtol`` and ``‖∇F(xᵏ)ᵀF(xᵏ) - ∇c(xᵏ)ᵀλᵏ‖ ≤ ϵtol * max(1, ‖λᵏ‖ / 100ncon)``.
```

Comment:
Is it intentional that it's εtol here and √εtol above?

Reply:
Yes, that's how it is in the code https://github.com/JuliaSmoothOptimizers/CaNNOLeS.jl/blob/7d2122a786a08905fa952494b9d8ee8675cdf6c6/src/CaNNOLeS.jl#L248